inferred-spans/src/test/java/co/elastic/otel/InferredSpansBackwardsCompatibilityConfigTest.java

```diff
@@ -0,0 +1,119 @@
+
+  @BeforeEach
+  @AfterEach
+  public void resetGlobalOtel() {
+    ProfilingActivationListener.ensureInitialized();
+    OtelReflectionUtils.shutdownAndResetGlobalOtel();
+  }
+
+  @Test
+  @DisabledOnOpenJ9
+  public void checkAllLegacyOptions(@TempDir Path tmpDir) {
+    String libDir = tmpDir.resolve("foo").resolve("bar").toString();
+    try (AutoConfigTestProperties props =
+        new AutoConfigTestProperties()
+            .put("elastic.otel.inferred.spans.enabled", "true")
+            .put("elastic.otel.inferred.spans.logging.enabled", "false")
+            .put("elastic.otel.inferred.spans.backup.diagnostic.files", "true")
+            .put("elastic.otel.inferred.spans.safe.mode", "16")
+            .put("elastic.otel.inferred.spans.post.processing.enabled", "false")
+            .put("elastic.otel.inferred.spans.sampling.interval", "7ms")
+            .put("elastic.otel.inferred.spans.min.duration", "2ms")
+            .put("elastic.otel.inferred.spans.included.classes", "foo*23,bar.baz")
+            .put("elastic.otel.inferred.spans.excluded.classes", "blub,test*.test2")
+            .put("elastic.otel.inferred.spans.interval", "2s")
+            .put("elastic.otel.inferred.spans.duration", "3s")
+            .put("elastic.otel.inferred.spans.lib.directory", libDir)) {
+
+      OpenTelemetry otel = GlobalOpenTelemetry.get();
+      List<SpanProcessor> processors = OtelReflectionUtils.getSpanProcessors(otel);
+      assertThat(processors).filteredOn(proc -> proc instanceof InferredSpansProcessor).hasSize(1);
+      InferredSpansProcessor processor =
+          (InferredSpansProcessor)
+              processors.stream()
+                  .filter(proc -> proc instanceof InferredSpansProcessor)
+                  .findFirst()
+                  .get();
+
+      InferredSpansConfiguration config = extractProfiler(processor).getConfig();
+      assertThat(config.isProfilingLoggingEnabled()).isFalse();
+      assertThat(config.isBackupDiagnosticFiles()).isTrue();
+      assertThat(config.getAsyncProfilerSafeMode()).isEqualTo(16);
+      assertThat(config.getSamplingInterval()).isEqualTo(Duration.ofMillis(7));
+      assertThat(wildcardsAsStrings(config.getIncludedClasses()))
+          .containsExactly("foo*23", "bar.baz");
+      assertThat(wildcardsAsStrings(config.getExcludedClasses()))
+          .containsExactly("blub", "test*.test2");
+      assertThat(config.getProfilingInterval()).isEqualTo(Duration.ofSeconds(2));
+      assertThat(config.getProfilingDuration()).isEqualTo(Duration.ofSeconds(3));
+      assertThat(config.getProfilerLibDirectory()).isEqualTo(libDir);
+    }
+  }
+
+  private SamplingProfiler extractProfiler(InferredSpansProcessor processor) {
+    try {
+      Field profilerField = processor.getClass().getDeclaredField("profiler");
+      profilerField.setAccessible(true);
+      return (SamplingProfiler) profilerField.get(processor);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
```

Comment:
[minor] wouldn't it be better to avoid using reflection and rely on a package-private helper class here ?